### PR TITLE
Relax XGBoost tolerance on GPU when comparing java and native predictions

### PR DESCRIPTION
**Pull request by michalkurka**
As demonstrated in https://github.com/h2oai/h2o-3/pull/5373/files sigmoid transformation
implemented on GPU gives different result than when running on CPU, therefore we pick a lower tolerance

This doesn't affect production results because native xgboost scoring
is only used in test. For all other scoring java scoring is used.

### CODE REVIEW

File: h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/XGBoostTest.java [hunk 1]
```diff
@@ -2420,4 +2420,29 @@ private void assertHasValidationError(hex.tree.xgboost.XGBoost builder, String f
     assertTrue(field, builder.validationErrors().contains(error));
   }
 
+  private void assertJavaScoring(XGBoostModel model, Frame testFrame, Frame preds) {
+    double rel_epsilon = Boolean.parseBoolean(confJavaPredict) ?
```
Comment (maurever) [inline h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/XGBoostTest.java:2424]:
```suggestion
    double relEpsilon = Boolean.parseBoolean(confJavaPredict) ? 
```

**Review by maurever — APPROVED** [review]
Just one nitpicking comment; otherwise looks good. Thank you, you found this issue @michalkurka!

File: h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/XGBoostTest.java (hunk 1 above)
Comment (michalkurka) [reply h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/XGBoostTest.java]: applied, thank you

**Comment by michalkurka** [comment]
thank you for checking @maurever

**Pull request merged**